introduction_to_applying_machine_learning/deepar_chicago_traffic_violations/deepar_chicago_traffic_violations.ipynb

```diff
@@ -0,0 +1,467 @@
+    "import boto3\n",
+    "import numpy as np\n",
+    "!conda install -y s3fs\n",
+    "np.random.seed(1)\n",
+    "import pandas as pd\n",
+    "import json\n",
+    "import matplotlib.pyplot as plt\n",
+    "import s3fs\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "---\n",
+    "\n",
+    "## Data\n",
+    "\n",
+    "\n",
+    "\n",
+    "The speed violations are captured by camera systems and available to improve the lives of public from the city of Chicago data portal. The Speed Camera Violation data set can be used to discern patterns in the data and gain meaningful insights.\n",
+    "\n",
+    "The data set contains multiple camera locations and daily violation counts. If we imagine that each daily violations for a camera as one time series, we can use Amazon SageMaker’s DeepAR algorithm to train a model for multiple streets simultaneously, and predict violation for multiple street cameras using the Amazon SageMaker’s DeepAR algorithm.\n",
+    "\n",
+    "The dataset contains several columns, we will use the Address, violation date, violations for the forecasting.\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "datafile = 'Chicago_Speed_Camera_Violations.csv'\n",
+    "speeding_violation_data_path = '{}/{}/{}.csv'.format(bucket,prefix, datafile)\n",
+    "s3_output_path = '{}/{}/output'.format(bucket, prefix)\n",
+    "s3_data_path = '{}/{}/data'.format(bucket, prefix)\n",
+    "train_data_path = '{}/{}/train/train.json'.format(bucket, prefix)\n",
+    "test_data_path = '{}/{}/test/test.json'.format(bucket, prefix)\n",
+    "\n",
+    "url = 'https://data.cityofchicago.org/api/views/hhkd-xvj4/rows.csv?accessType=DOWNLOAD'\n",
+    "\n",
+    "# get the data from City of Chicago site\n",
+    "r = requests.get(url, allow_redirects=True)\n",
+    "open(datafile, 'wb').write(r.content)\n",
+    "\n",
+    "# store it in S3 object\n",
+    "key = '{}/{}'.format(prefix, datafile)\n",
+    "s3client = boto3.client('s3', region_name=region)\n",
+    "with open(datafile, 'rb') as data:\n",
+    "    s3client.upload_fileobj(data, bucket, key)\n",
+    "\n",
+    "# get the object from S3, and display sample rows/columns\n",
+    "print(key)\n",
+    "print(bucket)\n",
+    "\n",
+    "obj = s3client.get_object(Bucket=bucket, Key=key)\n",
+    "pd.set_option('display.max_columns', 500)     \n",
+    "pd.set_option('display.max_rows', 50)    \n",
+    "df = pd.read_csv(io.BytesIO(obj['Body'].read()), encoding='utf-8')\n",
+    "# print first 10 lines to look at part of the dataset\n",
+    "df[['ADDRESS', 'VIOLATION DATE', 'VIOLATIONS']][0:10]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We convert the violation date from string format to date format, determine the range of violation dates, look at how many unique street address/camaera we have in our dataset.\n",
```

Comment:
[nitpick]
* add an "and" before "look"
* "address/camera" --> "addresses/cameras"